Week04/lesson08/homework/temp_converter/js/indexbonus4.js

```diff
@@ -0,0 +1,153 @@
+  var f = document.querySelector("#f").value;
+
+  if ((f === null || f === "") && (c === null || c === "")) {
+      document.querySelector("#alert").innerHTML = "Please fill one of the fields above.";
+      document.querySelector("#alert").style.border = "solid 1px red";
+      document.querySelector("#alert").style.backgroundColor = "white";
+      document.querySelector("#alert").style.padding = "5px";
+      document.querySelector("#c").style.borderColor = "red";
+      document.querySelector("#f").style.borderColor = "red";
+  }
+
+  else if ((f === null || f === "") && (c < 0)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "160px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://blogldc.s3.amazonaws.com/wp-content/uploads/2015/01/2015_01_16_coldphotos1.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((f === null || f === "") && (c < 10)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "120px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((f === null || f === "") && (c < 20)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "80px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((f === null || f === "") && (c < 30)){
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "40px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://www.vietnam-tour.biz/wp-content/uploads/2016/05/Dalat-and-Central-Highlands-seem-to-return-to-nice-weather-in-October.jpg')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if (f === null || f === "") {
+    f = c * 9 / 5 + 32;
+    document.querySelector("#f").value = Math.round(f*10)/10+"°F";
+    document.querySelector("#thermo").style.paddingTop = "0px";
+    document.querySelector(".wrapper").style.backgroundImage = "url('http://img.fotocommunity.com/eruption-0076bbc1-8b12-41ef-99ce-f4ea39b80229.jpg?height=400')";
+    document.querySelector(".wrapper").style.backgroundSize = "cover";
+  }
+
+  else if ((c === null || c === "") & (f < 32)) {
```

Comment:
Yes the `!` always means "not" so it's a double negative here: `!fNotFilledIn` means the `f` input is NOT not filled in (in other words, the `f` input *IS* filled in)